tcf_website/management/commands/load_grades.py

```diff
@@ -118,159 +179,135 @@ def load_row_into_dict(self, row):
                 total_weight += (
-                    course_instructor_grades[row][i] * grade_weights[i])
-
-            # calculate gpa without including other/drop/withdraw in
-            # total_enrolled
-            total_enrolled_filtered = total_enrolled - \
-                sum(course_instructor_grades[row][i] for i in (13, 14, 15))
-            if total_enrolled_filtered == 0:
-                total_enrolled_gpa = 0
-            else:
-                total_enrolled_gpa = total_weight / total_enrolled_filtered
-
-            course_instructor_grade_params = {
-                'subdepartment': row[0],
-                'number': row[1],
-                'first_name': row[2],
-                'middle_name': '',  # for backward compatibility
-                'last_name': row[3],
-                'email': row[4],
-                'course_id': self.courses.get(row[:2]),
-                'instructor_id': self.instructors.get(row[2:]),
-                'average': total_enrolled_gpa,
-                'a_plus': course_instructor_grades[row][0],
-                'a': course_instructor_grades[row][1],
-                'a_minus': course_instructor_grades[row][2],
-                'b_plus': course_instructor_grades[row][3],
-                'b': course_instructor_grades[row][4],
-                'b_minus': course_instructor_grades[row][5],
-                'c_plus': course_instructor_grades[row][6],
-                'c': course_instructor_grades[row][7],
-                'c_minus': course_instructor_grades[row][8],
-                'd_plus': course_instructor_grades[row][9],
-                'd': course_instructor_grades[row][10],
-                'd_minus': course_instructor_grades[row][11],
-                'f': course_instructor_grades[row][12],
-                'ot': course_instructor_grades[row][13],  # other/pass
-                'drop': course_instructor_grades[row][14],
-                'withdraw': course_instructor_grades[row][15],
-                'total_enrolled': total_enrolled
-            }
+                    self.course_instructor_grades[row][i] * grade_weights[i])
+
+            course_instructor_grade_params = self.set_grade_params(
+                row, total_enrolled, total_weight, has_instructor=True)
             unsaved_cig_instance = CourseInstructorGrade(
                 **course_instructor_grade_params)
             unsaved_cig_instances.append(unsaved_cig_instance)
         CourseInstructorGrade.objects.bulk_create(unsaved_cig_instances)
         if self.verbosity > 0:
             print('Done creating CourseInstructorGrade instances')
+
+        if self.log_missing_instructors:
+            print('Writing missing instructors to missing-instructors.txt')
+            with open('missing-instructors.txt', 'w') as file:
+                for instructor in self.missing_instructors:
+                    file.write(instructor)
+
+    def set_grade_params(self, row, total_enrolled, total_weight, has_instructor):
```

Comment:
`has_instructor` is a bit of a misleading parameter that can be confused with the missing instructors list. Instead maybe `is_instructor_grade` or something like that?

Reply:
Resolved